x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java

```diff
@@ -113,6 +116,14 @@ public void testFullClusterRestart() throws Exception {
             logger.info("Restarting all nodes");
             internalCluster().fullRestart();
             logger.info("Restarted all nodes");
+            // We should wait for the indices to be available until we spin wait for persistent tasks to be assigned
+            logger.info("Waiting for ML indices to become available.");
+            ensureYellow(
+                AnomalyDetectorsIndex.jobStateIndexPattern(),
+                MlMetaIndex.INDEX_NAME,
+                AnomalyDetectorsIndex.configIndexName(),
+                AnomalyDetectorsIndex.jobResultsIndexPrefix() + AnomalyDetectorsIndexFields.RESULTS_INDEX_DEFAULT);
+            logger.info("ML indices are now available.");
```

Comment:
The pattern of the other tests that use `run()` is to call `ensureStableClusterOnAllNodes()` at the end of it.  Does that work here?  I.e., replace your new lines 119-126 with `ensureStableClusterOnAllNodes(3);`.

Reply:
Hmmm, let me take a gander.